src/nvim/api/ui.c

```diff
@@ -180,6 +183,42 @@ void ui_attach(uint64_t channel_id, Integer width, Integer height,
   api_free_dictionary(opts);
 }
 
+void nvim_set_terminfo(uint64_t channel_id, Dictionary options, Error *error)
+  FUNC_API_SINCE(6) FUNC_API_REMOTE_ONLY
+{
+  if (!pmap_has(uint64_t)(connected_uis, channel_id)) {
+    api_set_error(error, kErrorTypeException,
+                  "UI not attached to channel: %" PRId64, channel_id);
+    return;
+  }
+
+  for (size_t i = 0; i < options.size; i++) {
+    if (!strcmp(options.items[i].key.data, "termname")) {
+      set_tty_option("term", xstrdup(options.items[i].value.data.string.data));
+    } else if (!strcmp(options.items[i].key.data, "t_Co")) {
+      t_colors = (int)options.items[i].value.data.integer;
+    } else if (!strcmp(options.items[i].key.data, "bg")) {
+      set_option_value("bg",
+                      options.items[i].value.data.array.items[0].data.integer,
+                      options.items[i].value.data.array.items[1].data.string.data,
+                      (int)options.items[i].value.data.array.items[3].data.integer);
```

Comment:
This 3 should be a 2 I guess. But this array is unnecessary, you can just send "light" or "dark" string, the rest are constants.